src/utils/d2c_messaging/src/d2c_messaging.c

```diff
@@ -456,7 +491,7 @@ void ADUC_D2C_Messaging_Uninit()
     if (s_core_initialized)
     {
         // Cancel pending messages
-        for (int i = 0; i < ADUC_D2C_Message_Type_Max; i++)
+        for (int i = 0; i < ADUC_D2C_Message_Type_Max && s_messageProcessingContext[i].initialized; i++)
         {
             pthread_mutex_lock(&s_messageProcessingContext[i].mutex);
             if (s_pendingMessageStore[i].content != NULL)
```

Comment:
I do not see where s_pendingMessageStore[i].content is set? Basically, it seems this will always be NULL (and if-stmt always will be false).  Can it be removed (along with its mutex)?

Reply:
Maybe larger scope than this PR, but having mutex per s_messagePorcessingContext element (ADUC_D2C_Message_Type)seems excessive since there is only one worker thread processing these. We could increase in the future, but I doubt there will be multiple worker threads for any one type so probably sufficient to just have a single read-writer lock for accessing all the types.